pkg/provisioners/gcppubsub/dispatcher/receiver/receiver.go

```diff
@@ -76,13 +80,21 @@ func (r *Receiver) sendEventToTopic(channel provisioners.ChannelReference, messa
 		return errors.New("status.internal is blank")
 	}
 
-	psr, err := r.createPubSubReceiver(ctx, pcs)
-	if err != nil {
-		logging.FromContext(ctx).Info("Unable to create pubSubReceiver", zap.Error(err))
-		return err
+	cached := r.cache.Get(pcs.Topic)
+	if cached == nil {
+		cached, err = r.createPubSubReceiver(ctx, pcs)
+		if err != nil {
+			logging.FromContext(ctx).Info("Unable to create pubSubReceiver", zap.Error(err))
+			return err
+		}
 	}
 
-	result := psr.Publish(ctx, &pubsub.Message{
+	psr, ok := cached.(*pubSubReceiver)
+	if !ok {
+		return fmt.Errorf("unexpected type in the cache[%s] - %T", pcs.Topic, cached)
```

Comment:
should we remove the violating entry from the cache here then?

Reply:
Done.